app/models/piece.rb

```diff
@@ -2,12 +2,75 @@ class Piece < ApplicationRecord
   belongs_to :game
   delegate :king, :queen, :bishop, :knight, :rook, :pawn, to: :pieces
 
-  #needs models that inherit from piece called pawn, rook, knight, bishop, queen, king
-
   def outside_board?(position_x, position_y)
     position_x < 0 || position_x > 7 || position_y < 0 || position_y > 7
   end 
 
+  def white? 
+    self.color == 'white'
+  end 
+
+  def black? 
+    self.color == 'black'
+  end 
+
+  def rook?
+    self.type == 'Rook'
+  end 
+
+  def bishop?
+    self.type == 'Bishop'
+  end 
+
+  def king?
+    self.type == 'King'
+  end 
+
+  def queen?
+    self.type == 'Queen'
+  end 
+
+  def knight?
+    self.type == 'Knight'
+  end 
+
+  def pawn?
+    self.type == 'Pawn'
+  end 
+
+
+  def show_image
```

Comment:
Nice, @snsmith92, we've got rid of the db for images. Let's try to reduce it even more.. the images already have a consistent naming, like: `knight-black.png`... if you would try to explain the naming concept of image naming, does it follow a pattern? can we use this pattern to have the `show_image` method be just one line? 🤔 Hint: (http://rubymonk.com/learning/books/1-ruby-primer/chapters/5-strings/lessons/31-string-basics)[String interpolation]

Reply:
done!